src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -97,6 +97,37 @@ public static SourceText From(string text, Encoding encoding = null, SourceHashA
             return new StringText(text, encoding, checksumAlgorithm: checksumAlgorithm);
         }
 
+#pragma warning disable RS0026 // Do not add multiple public overloads with optional parameters
```

Comment:
I think TextReader is morally equivalent to string here: you get chars not bytes out of it. As such, it is OK to defer making the embedded blob and to do so by re-encoding it using the given encoding. It would be good to add a FromSource_TextReader test to EmbeddedTextTests. It should work as well as passing the same text to the string overload.

---
In reply to: [83501039](https://github.com/dotnet/roslyn/pull/14531#discussion_r83501039) [](ancestors = 83501039)